### ISSUE TITLE
wwsympa.service missing in debian package

### ISSUE THREAD

**Issue by Bothwalker** [sympa-community/sympa#765]
Hello, i followed the guide to install sympa here: https://sympa-community.github.io/manual/install.html
My OS is Debian 10. When i have to install the wwsympa.service (https://sympa-community.github.io/manual/install/configure-http-server-spawnfcgi.html) the file is missing. 
Can anyone help me with this step?

**Comment by ldidry** [comment]
ping @racke

**Comment by ikedas** [comment]
I [added a note](sympa-community/sympa-community.github.io@856c495) that currently Debian won't use spawn-fcgi.  Please see [updated documentation](https://sympa-community.github.io/manual/install/configure-http-server-spawnfcgi.html).

**Comment by ludodo** [comment]
Hello,
Thank's for adding documentation, but I must admit I'm not sure what to do and I'm unable to make it work. Do we have to use systemd to make the service run ? I tried to but :
- wwsympa.socket does not launch wwsympaservice,
- also I manually run wwsympa.service, and process list show me it runs, I still don't have the wws working

Please, a little help would be appreciate

**Comment by racke** [comment]
@ludodo You don't have to run it with systemd, but it is certainly recommended.

Can you please post the systemd unit files which you are using?

**Comment by ludodo** [comment]
Here my wwsympa.service file : 
[Unit]
Description = WWSympa - Web interface for Sympa mailing list manager (service)
After = syslog.target sympa.service

[Service]
Type = simple
User = sympa
Group = sympa
ExecStart = /usr/lib/cgi-bin/sympa/wwsympa.fcgi
StandardOutput = null
StandardInput = socket
StandardError = null
Restart=on-failure
Envionnment="FCGI_OPTS=-M 0600 -U www-data"

[Install]
WantedBy = multi-user.target

**Comment by ludodo** [comment]
And the wwsympa.socket file
[Unit]
Description = WWSympa - Web interface for Sympa mailing list manager (socket)

[Socket]
SocketUser = www-data
SocketMode = 0600
ListenStream = /run/sympa/wwsympa.socket

[Install]
WantedBy = sockets.target

**Comment by racke** [comment]
This is not needed (spawn-fcgi options), but probably ignored by systemd because of the
typos in the _Environment_ keyword.

`Envionnment="FCGI_OPTS=-M 0600 -U www-data"`


Try for the Socket section:
```

[Socket]
SocketUser = www-data
SocketGroup = www-data
SocketMode = 0660
ListenStream = /run/sympa/wwsympa.socket
```

**Comment by ludodo** [comment]
Thanks for your help.
I changed the 2 lines for the socket. Socket service start, but still does not start wwsympa.service
I can still start wwsympa.service "by hand", and it runs as you can see :
www-data  1609  1606  0 12:00 ?        00:00:00 /usr/sbin/apache2 -k start
postfix   1610   657  0 Dec04 ?        00:00:00 tlsmgr -l -t unix -u -c
www-data  1613  1606  0 12:00 ?        00:00:00 /usr/sbin/apache2 -k start
sympa     1951     1  0 12:32 ?        00:00:00 /usr/bin/perl /usr/lib/cgi-bin/sympa/wwsympa.fcgi
sympa    31075     1  0 00:00 ?        00:00:04 /usr/bin/perl /usr/lib/sympa/bin/task_manager.pl
sympa    31076     1  0 00:00 ?        00:00:07 /usr/bin/perl /usr/lib/sympa/bin/bulk.pl
sympa    31078     1  0 00:00 ?        00:00:01 /usr/bin/perl /usr/lib/sympa/bin/archived.pl
sympa    31080     1  0 00:00 ?        00:00:01 /usr/bin/perl /usr/lib/sympa/bin/sympa_msg.pl
sympa    31081     1  0 00:00 ?        00:00:01 /usr/bin/perl /usr/lib/sympa/bin/bounced.pl

But why my web-browser download a sympa file when putting the URL : http://mydomain/sympa instead of running the script ?

**Comment by racke** [comment]
The problem might be your web server configuration, please post the relevant parts. Also make sure that the URL matches _wwsympa_url_ in Sympa configuration and domain in the web server configuration.

**Comment by ludodo** [comment]
OK, here's apache conf :
<VirtualHost *:80>
    ServerName listes.mondomaine.org
    ServerAdmin postmaster@listes.mondomaine.org

    LoadModule alias_module modules/mod_alias.so
    LoadModule proxy_module modules/mod_proxy.so
    LoadModule proxy_fcgi_module modules/mod_proxy_fcgi.so

    <Location /sympa>
        SetHandler "proxy:unix:/run/sympa/wwsympa.socket|fcgi://localhost/"
        Require all granted
    </Location>

    <location /static-sympa>
        Require all granted
    </location>

    Alias /static-sympa /usr/share/sympa/static_content

</VirtualHost>

**Comment by ludodo** [comment]
and sympa.conf : 
###\\\\ Web interface parameters ////###

## URL prefix of web interface
## This is used to construct URLs of web interface.
# was wwsympa_url http://listes.cen-centrevaldeloire.org/sympa
wwsympa_url https://listes.cen-centrevaldeloire.org/sympa

## URL for static contents
## HTTP server have to map it with "static_content_path" directory.
static_content_url  /static-sympa

## Directory for static contents
static_content_path /usr/share/sympa/static_content

**Comment by racke** [comment]
So if you use https://listes.mondomaine.org/sympa in your browser it doesn't use the configuration in your virtual host config (port 80).

**Comment by ludodo** [comment]
hum, it's because there's e nginx proxy in front of the sympa container with apache. Nginx forward to http (80)

**Comment by ludodo** [comment]
for the moment, I test directly whith http://local_serverip/sympa

**Comment by racke** [comment]
That doesn't work as URL needs to match wwsympa_url.

**Comment by ludodo** [comment]
ok, I understand. If I use the wwsympa_url witch pass-through the nginx-proxy to the apache server listening on http, I've got the same result : donwloading sympa file . I can post here the nginx conf of the reverse proxy that I use for numerous other webapps :
server {
    listen 0.0.0.0:80;
    server_name listes.mondomaine.org;
    access_log /var/log/nginx/sympa_access.log;
    error_log /var/log/nginx/sympa_error.log emerg;
    rewrite     ^   https://$host$request_uri? permanent;
}
server {
    listen 443 ssl;
    server_name listes.mondomaine.org;
    access_log /var/log/nginx/sympa_access.log;
    error_log /var/log/nginx/sympa_error.log error;

    #SSL Certificates
    ssl on;
    ssl_certificate "/etc/letsencrypt/live/listes.mondomaine.org/cert.pem";
    ssl_certificate_key "/etc/letsencrypt/live/listes.mondomaine.org/privkey.pem";
    ssl_protocols SSLv3 TLSv1 TLSv1.1 TLSv1.2;
    ssl_session_timeout 10m;
    ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
    ssl_prefer_server_ciphers  on;
#    include proxy.conf;

   location / {
         proxy_pass http://192.168.100.114/;
         }
    location ~ \.php$ {
         proxy_set_header X-Real-IP $remote_addr;
         proxy_set_header X-Forwarded-For $remote_addr;
         proxy_set_header Host $host;
         proxy_pass http://192.168.100.114/$request_uri;
        }
    location ~ /\.ht {
     deny all;
    }
}

**Comment by racke** [comment]
My guess would be that the problem is the Apache configuration in the container (why don't you use Nginx here as well?).
So my suggestion is to set _log_level_ to 1, which would show how wwsympa sees the incoming information.

If it works, it should look like that:

```
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: ORIG_PATH_INFO=
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: ORIG_SCRIPT_NAME=/sympa/info/testlist
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: PATH_INFO=/info/testlist
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: QUERY_STRING=
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: REMOTE_ADDR=10.0.2.2
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: REMOTE_HOST=
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: REQUEST_METHOD=GET
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: SCRIPT_NAME=/sympa
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: SERVER_NAME=buster-sympa.icdev.de
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: SERVER_PORT=80
Dec 07 14:23:33 buster wwsympa[25985]: debug main:: SYMPA_DOMAIN=buster-sympa.icdev.de

```

**Comment by ludodo** [comment]
I made the same conclusion and tried nginx -> same result.
So I looked at the nginx proxy and the mismatch URL with the passtrough, and found the two solution :
- change in the reverse proxy the proxy-pass to http://192.168.100.114/sympa
- or change in the sympa container the web server and sympa configuration to '/' instead of '/sympa'

And now it works.
Thank's a lot for your help and the time passed for us.
